plugins/module_utils/network/sonic/config/qos_maps/qos_maps.py

```diff
@@ -0,0 +1,564 @@
+                name = m.get('name')
+                entries = m.get('entries')
+
+                for cfg_m in cfg_maps:
+                    map_dict = {}
+                    cfg_name = cfg_m.get('name')
+                    cfg_entries = cfg_m.get('entries')
+
+                    if name and name == cfg_name:
+                        oc_map = lookup_dict['oc_map']
+                        if not entries:
+                            url = '%s/openconfig-qos-maps-ext:%ss/%s=%s' % (QOS_PATH, oc_map, oc_map, name)
+                            requests.append({'path': url, 'method': DELETE})
+                            map_dict.update({'name': name})
+
+                        else:
+                            if entries and cfg_entries:
+                                entries_list = []
+                                for entry in entries:
+                                    entry_dict = {}
+                                    attr1_key = lookup_dict['attr1']
+                                    attr2_key = lookup_dict['attr2']
+                                    attr1 = entry.get(attr1_key)
+                                    attr2 = entry.get(attr2_key)
+
+                                    for cfg_entry in cfg_entries:
+                                        cfg_attr1 = cfg_entry.get(attr1_key)
+                                        cfg_attr2 = cfg_entry.get(attr2_key)
+
+                                        if attr1 is not None and attr1 == cfg_attr1:
+                                            oc_attr2 = lookup_dict['oc_attr2']
+                                            if attr2 is not None and attr2 == cfg_attr2:
+                                                url = '%s/openconfig-qos-maps-ext:%ss/%s=%s/' % (QOS_PATH, oc_map, oc_map, name)
+                                                url += '%s-entries/%s-entry=%s/config/%s' % (oc_map, oc_map, attr1, oc_attr2)
+                                                requests.append({'path': url, 'method': DELETE})
+                                                entry_dict.update({attr1_key: attr1, attr2_key: attr2})
+                                            if attr2 is None:
+                                                url = '%s/openconfig-qos-maps-ext:%ss/%s=%s/' % (QOS_PATH, oc_map, oc_map, name)
+                                                url += '%s-entries/%s-entry=%s' % (oc_map, oc_map, attr1)
+                                                requests.append({'path': url, 'method': DELETE})
+                                                entry_dict.update({attr1_key: attr1})
+                                            if entry_dict:
+                                                entries_list.append(entry_dict)
+
+                                if entries_list:
+                                    map_dict.update({'name': name, 'entries': entries_list})
+
+                    if map_dict:
+                        maps_list.append(map_dict)
```

Comment:
After handling a matching map in the configuration, there should be a 'break' to avoid checking the remaining configuration maps for a match.